src/views/Login.tsx

```diff
@@ -1,57 +1,97 @@
-import React from 'react'
+import React, { useState } from 'react'
 import '../assets/scss/Login.scss'
-import { Controller, useForm } from 'react-hook-form'
-import TextField from '@mui/material/TextField'
+import { useForm } from 'react-hook-form'
 import Button from '@mui/material/Button'
+import axios from 'axios'
+import { useAtom } from 'jotai'
+import tokenAtom from '../atoms/token'
+import PasswordField from '../components/PasswordField'
+import UsernameField from '../components/UsernameField'
 
 export default function Login (): JSX.Element {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [token, setToken] = useAtom(tokenAtom)
+  const [errorMessage, setErrorMessage] = useState('')
+  const [showErrorMessage, setShowErrorMessage] = useState(false)
   const { handleSubmit, control } = useForm()
-  // TODO: connect onSubmit to login endpoint
-  const onSubmit = (data: any): void => console.log(data)
+
+  const onLoginSubmit = async (data: any): Promise<void> => {
+    await validateLogin(data.username, data.password)
+  }
+
+  const onRegisterSubmit = async (data: any): Promise<void> => {
+    await registerUser(data.username, data.password)
+  }
+
+  const validateLogin = async (username: string, password: string): Promise<void> => {
+    try {
+      const res = await axios.post('/auth/login', { username, password })
+
+      if (res.status === 200) {
+        if (res.data.token) {
+          setToken(res.data.token)
+        }
+        console.log(res.data.message)
+      }
+    } catch (err) {
+      if (axios.isAxiosError(err)) {
+        if (err.response !== undefined) {
+          if (err.response.data.message === 'Invalid username or password') {
+            setErrorMessage(err.response.data.message)
+            setShowErrorMessage(true)
+          }
+        }
+      }
+    }
+  }
+
+  const registerUser = async (username: string, password: string): Promise<void> => {
+    try {
+      const res = await axios.post('/register/add-user', { username, password })
+
+      if (res.status === 200) {
+        handleSubmit(onLoginSubmit)
+        console.log(res.data.message)
+      }
```

Comment:
`res.status` will not be anything other than 200 per API spec, provided the request didn't error. Also errant log again it seems.
```suggestion
      handleSubmit(onLoginSubmit)
```